server/libs/domain/src/storage-template/storage-template.service.ts

```diff
@@ -52,24 +53,29 @@ export class StorageTemplateService {
   async handleTemplateMigration() {
     try {
       console.time('migrating-time');
-      const assets = await this.assetRepository.getAll();
+
+      const assetPagination = usePagination(JOBS_ASSET_PAGINATION_SIZE, (pagination) =>
+        this.assetRepository.getAll(pagination),
+      );
       const users = await this.userRepository.getList();
 
-      const livePhotoMap: Record<string, AssetEntity> = {};
+      for await (const assets of assetPagination) {
+        const livePhotoMap: Record<string, AssetEntity> = {};
 
-      for (const asset of assets) {
-        if (asset.livePhotoVideoId) {
-          livePhotoMap[asset.livePhotoVideoId] = asset;
+        for (const asset of assets) {
+          if (asset.livePhotoVideoId) {
+            livePhotoMap[asset.livePhotoVideoId] = asset;
+          }
         }
```

Comment:
This code doesn't work. It's trying to find the parent photo asset from the (current) child motion asset. Unless they are both in the same 1000 items, this will always be null. However, this is all to ensure that both files have the same name. I believe this is now fixed at time of upload [here](https://github.com/immich-app/immich/blob/main/server/apps/immich/src/api-v1/asset/asset.service.ts#L107), so we might be fine removing it anyway.

Reply:
I'm fine removing it :+1:

Reply:
BTW - the filename input isn't even used anymore :-)

https://github.com/immich-app/immich/pull/2517